Migrate "Improve > International > Locations > Zones" and create/edit of zones

ks129 opened the pull request:
<!-----------------------------------------------------------------------------
Thank you for contributing to the PrestaShop project! 

Please take the time to edit the "Answers" rows below with the necessary information.

Check out our contribution guidelines to find out how to complete it:
https://devdocs.prestashop.com/1.7/contribute/contribution-guidelines/#pull-requests
------------------------------------------------------------------------------>

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Migrated "Improve > International > Locations > Zones" index and it's forms (also all endpoints) to new modern Symfony controller.
| Type?         | refacto
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #10603 and fixes #10608.
| How to test?  | "Improve > International > Locations > Zones" should use new controller (also all other features, like enabling, deleting, bulk actions, create, edit)

<!-- Click the form's "Preview" button to make sure the table is functional in GitHub. Thank you! -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/20370)
<!-- Reviewable:end -->

ks129 committed:
Create zone adding command, handler and it's interface

ks129 committed:
Create zone editing command, handler, it's interface and result

ks129 committed:
Create zone for editing query, handler and it's interface

ks129 committed:
Add ZoneId class for commands and handlers

ks129 committed:
Create base Zone exception

ks129 committed:
Add Zone not found exception

ks129 committed:
Add Zone exception for case when invalid values provided

ks129 committed:
Add Zone exception for case when some values are missing

ks129 committed:
Create all Zones grid parts

ks129 committed:
Create Zone exception for case when (bulk) delete fails

ks129 committed:
Create Zone deletion command, handler and it's interface

ks129 committed:
Create Zone bulk deletion command, handler and it's interface

ks129 committed:
Create Zone bulk toggling command, handler and it's interface

ks129 committed:
Create Zone editing and toggling exceptions

ks129 committed:
Create Zone status toggling command, handler and it's interface

ks129 committed:
Create services for all zone command handlers

ks129 committed:
Create all assets for zone form

ks129 committed:
Create views for form grid (index), create and edit

ks129 committed:
Add JS for zones grid

ks129 committed:
Create new modern Zone controller and routing

ks129 committed:
Remove old legacy zones controller

LouiseBonnard commented on src/Core/Grid/Definition/Factory/ZoneGridDefinitionFactory.php:
```diff
@@ -0,0 +1,240 @@
+     */
+    protected function getName()
+    {
+        return $this->trans('Zones', [], 'Admin.Global');
```
Hi there, please localize it in _Admin.International.Feature_ or _Admin.Navigation.Menu_, thanks!

LouiseBonnard commented on src/Core/Grid/Definition/Factory/ZoneGridDefinitionFactory.php:
```diff
@@ -0,0 +1,240 @@
+            )
+            ->add(
+                (new ActionColumn('actions'))
+                    ->setName($this->trans('Actions', [], 'Admin.Actions'))
```
```suggestion
                    ->setName($this->trans('Actions', [], 'Admin.Global'))
```

LouiseBonnard commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,381 @@
+            $result = $formHandler->handleFor($zoneId, $zoneForm);
+
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Successful update.', 'Admin.Notifications.Success'));
```
```suggestion
                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
```

LouiseBonnard commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,381 @@
+    {
+        return [
+            CannotEditZoneException::class => $this->trans(
+                'An error occurred while editing zone.',
```
```suggestion
                'An error occurred while editing the zone.',
```

matks commented:
@ks129 you're very motivated 😄

LouiseBonnard commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,381 @@
+                'Admin.International.Notification'
+            ),
+            InvalidZoneValuesException::class => $this->trans(
+                'Invalid value(s) provided for zone.',
```
What invalid values can be provided?

ks129 replied:
Actually there can't be any. Only is empty name but this is handled by another exception. Removed this exception.

LouiseBonnard replied:
Alright, thanks a lot!

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Zone/Blocks/form.html.twig:
```diff
@@ -0,0 +1,54 @@
+  <div class="card-footer">
+    <div class="d-inline-flex">
+      <a href="{{ path('admin_zones_index') }}" class="btn btn-outline-secondary">
+        {{ 'Cancel'|trans({}, 'Admin.Global') }}
```
```suggestion
        {{ 'Cancel'|trans({}, 'Admin.Actions') }}
```

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Zone/Blocks/form.html.twig:
```diff
@@ -0,0 +1,54 @@
+      </a>
+    </div>
+    <div class="d-inline-flex float-right">
+      <button class="btn btn-primary" id="save-button">{{ 'Save'|trans({}, 'Admin.Global') }}</button>
```
```suggestion
      <button class="btn btn-primary" id="save-button">{{ 'Save'|trans({}, 'Admin.Actions') }}</button>
```

LouiseBonnard commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Zone/Blocks/form.html.twig:
```diff
@@ -0,0 +1,54 @@
+{{ form_start(zoneForm) }}
+<div class="card">
+  <div class="card-header">
+    {{ 'Zones'|trans({}, 'Admin.Global') }}
```
Same as the first comment: to be localized in _Admin.International.Feature_ or _Admin.Navigation.Menu_ depending on what option you picked. ;-)

ks129 committed:
Fix translation domain on zones grid name

ks129 committed:
Fix translation domain on Zone form header

ks129 committed:
Replace zone form save and cancel button translation domain

Replace domain from `Admin.Global` to `Admin.Actions`.

Co-authored-by: LouiseBonnard <32565890+LouiseBonnard@users.noreply.github.com>

ks129 committed:
Change zone grid action row translation domain

ks129 committed:
Update zone controller edit success message

ks129 committed:
Fix zone controller edit error message

ks129 committed:
Remove zone invalid values handling

It's impossible to pass invalid values to zone, so this exception
raising and catching is useless.

ks129 committed:
Remove zones legacy controller Ajax test

Progi1984 commented on src/Core/Form/IdentifiableObject/DataHandler/ZoneFormDataHandler.php:
```diff
@@ -0,0 +1,79 @@
+     *
+     * @param array $data
+     *
+     * @return mixed
```
```suggestion
     * @return int
```

Progi1984 commented on src/Core/Form/IdentifiableObject/DataProvider/ZoneFormDataProvider.php:
```diff
@@ -0,0 +1,81 @@
+     *
+     * @param int $id
+     *
+     * @return mixed
```
```suggestion
     * @return array
```

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,376 @@
+     */
+    public function bulkToggleStatus(string $status, Request $request)
+    {
+        $status = $status == 'enable';
```
```suggestion
        $status = $status === 'enable';
```

Progi1984 reviewed (commented):
Some small comments ;) And it's :ok_hand: for me

ks129 committed:
Add missing = to zone status toggling comparisation

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Fix PHPDoc return types

Co-authored-by: Progi1984 <progi1984@gmail.com>

PierreRambaud commented on src/Adapter/Zone/CommandHandler/BulkDeleteZoneHandler.php:
```diff
@@ -0,0 +1,59 @@
+            $zone = new Zone($zoneId->getValue());
+
+            if (0 >= $zone->id) {
+                throw new ZoneNotFoundException(sprintf('Unable to find zone with id "%s" for deletion', $zoneId->getValue()));
```
```suggestion
                throw new ZoneNotFoundException(sprintf('Unable to find zone with id "%d" for deletion', $zoneId->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/BulkDeleteZoneHandler.php:
```diff
@@ -0,0 +1,59 @@
+            }
+
+            if (!$zone->delete()) {
+                throw new DeleteZoneException(sprintf('An error occurred when deleting zone with id "%s"', $zoneId->getValue()), DeleteZoneException::FAILED_BULK_DELETE);
```
```suggestion
                throw new DeleteZoneException(sprintf('An error occurred when deleting zone with id "%d"', $zoneId->getValue()), DeleteZoneException::FAILED_BULK_DELETE);
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/BulkToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,67 @@
+            $zone = new Zone($zoneId->getValue());
+
+            if (0 >= $zone->id) {
+                throw new ZoneNotFoundException(sprintf('Zone object with id "%s" has been not found for status changing', $command->getZoneId()->getValue()));
```
```suggestion
                throw new ZoneNotFoundException(sprintf('Zone object with id "%d" has been not found for status changing', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/BulkToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,67 @@
+
+            try {
+                if (!$zone->save()) {
+                    throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status for zone with id "%s"', $zoneId->getValue()));
```
```suggestion
                    throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status for zone with id "%d"', $zoneId->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/BulkToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,67 @@
+                    throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status for zone with id "%s"', $zoneId->getValue()));
+                }
+            } catch (PrestaShopException $e) {
+                throw new ZoneException(sprintf('An error occurred while updating zone status with id "%s"', $zoneId->getValue()));
```
```suggestion
                throw new ZoneException(sprintf('An error occurred while updating zone status with id "%d"', $zoneId->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/DeleteZoneHandler.php:
```diff
@@ -0,0 +1,57 @@
+        $zone = new Zone($command->getZoneId()->getValue());
+
+        if (0 >= $zone->id) {
+            throw new ZoneNotFoundException(sprintf('Unable to find zone with id "%s" for deletion', $command->getZoneId()->getValue()));
```
```suggestion
            throw new ZoneNotFoundException(sprintf('Unable to find zone with id "%d" for deletion', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/DeleteZoneHandler.php:
```diff
@@ -0,0 +1,57 @@
+        }
+
+        if (!$zone->delete()) {
+            throw new DeleteZoneException(sprintf('Cannot delete zone with id "%s"', $command->getZoneId()->getValue()), DeleteZoneException::FAILED_DELETE);
```
```suggestion
            throw new DeleteZoneException(sprintf('Cannot delete zone with id "%d"', $command->getZoneId()->getValue()), DeleteZoneException::FAILED_DELETE);
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/EditZoneHandler.php:
```diff
@@ -0,0 +1,74 @@
+        try {
+            $zone = new Zone($command->getZoneId()->getValue());
+        } catch (PrestaShopException $e) {
+            throw new ZoneException(sprintf('Failed to get zone with id "%s"', $command->getZoneId()->getValue()), 0, $e);
```
```suggestion
            throw new ZoneException(sprintf('Failed to get zone with id "%d"', $command->getZoneId()->getValue()), 0, $e);
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/EditZoneHandler.php:
```diff
@@ -0,0 +1,74 @@
+        }
+
+        if ($zone->id !== $command->getZoneId()->getValue()) {
+            throw new ZoneNotFoundException(sprintf('Zone with id "%s" was not found', $command->getZoneId()->getValue()));
```
```suggestion
            throw new ZoneNotFoundException(sprintf('Zone with id "%d" was not found', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/EditZoneHandler.php:
```diff
@@ -0,0 +1,74 @@
+
+        try {
+            if (!$zone->update()) {
+                throw new CannotEditZoneException(sprintf('Cannot update zone with id "%s"', $zone->id));
```
```suggestion
                throw new CannotEditZoneException(sprintf('Cannot update zone with id "%d"', $zone->id));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/EditZoneHandler.php:
```diff
@@ -0,0 +1,74 @@
+                throw new CannotEditZoneException(sprintf('Cannot update zone with id "%s"', $zone->id));
+            }
+        } catch (PrestaShopException $e) {
+            throw new CannotEditZoneException(sprintf('Cannot update zone with id "%s"', $zone->id));
```
```suggestion
            throw new CannotEditZoneException(sprintf('Cannot update zone with id "%s"', $zone->id));
```
```suggestion
            throw new CannotEditZoneException(sprintf('Cannot update zone with id "%d"', $zone->id));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/ToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,62 @@
+            $zone = new Zone($command->getZoneId()->getValue());
+
+            if (0 >= $zone->id) {
+                throw new ZoneNotFoundException(sprintf('Zone object with id "%s" has been not found for status changing', $command->getZoneId()->getValue()));
```
```suggestion
                throw new ZoneNotFoundException(sprintf('Zone object with id "%d" has been not found for status changing', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/ToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,62 @@
+            }
+
+            if (false === $zone->toggleStatus()) {
+                throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status of zone with id "%s"', $command->getZoneId()->getValue()));
```
```suggestion
                throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status of zone with id "%d"', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/CommandHandler/ToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,62 @@
+                throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status of zone with id "%s"', $command->getZoneId()->getValue()));
+            }
+        } catch (PrestaShopException $e) {
+            throw new ZoneException(sprintf('An error occurred when toggling status for zone with id "%s"', $command->getZoneId()->getValue()));
```
```suggestion
            throw new ZoneException(sprintf('An error occurred when toggling status for zone with id "%d"', $command->getZoneId()->getValue()));
```

PierreRambaud commented on src/Adapter/Zone/QueryHandler/GetZoneForEditingHandler.php:
```diff
@@ -0,0 +1,58 @@
+        $zone = new Zone($zoneId->getValue());
+
+        if ($zone->id !== $zoneId->getValue()) {
+            throw new ZoneNotFoundException(sprintf('Zone with id "%s" not found', $zoneId->getValue()));
```
```suggestion
            throw new ZoneNotFoundException(sprintf('Zone with id "%d" not found', $zoneId->getValue()));
```

PierreRambaud commented on src/Core/Domain/Zone/ValueObject/ZoneId.php:
```diff
@@ -0,0 +1,69 @@
+    private function assertIntegerIsGreaterThanZero(int $zoneId)
+    {
+        if (0 > $zoneId) {
+            throw new ZoneException(sprintf('Zone id %s is invalid. Zone id have to be number bigger than zero.', $zoneId));
```
```suggestion
            throw new ZoneException(sprintf('Zone id %d is invalid. Zone id have to be number bigger than zero.', $zoneId));
```

PierreRambaud commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,376 @@
+        $zoneGridFactory = $this->get('prestashop.core.grid.factory.zone');
+        $zoneGrid = $zoneGridFactory->getGrid($zoneFilters);
+
+        return $this->render('PrestaShopBundle:Admin/Improve/International/Zone:index.html.twig', [
```
```suggestion
        return $this->render('@Prestashop/Admin/Improve/International/Zone:index.html.twig', [
```
:thinking: ?

ks129 replied:
Sorry, PhpStorm autocomplete.

PierreRambaud commented on src/PrestaShopBundle/Resources/config/routing/admin/improve/international/zone.yml:
```diff
@@ -0,0 +1,77 @@
+    _legacy_link: AdminZones
+
+admin_zones_search:
+  path: /
```
```suggestion
  path: /search
```

ks129 replied:
Why? All others have this as route `/` as POST.

PierreRambaud replied:
Why did they do this :scream:

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Zone/Blocks/form.html.twig:
```diff
@@ -0,0 +1,54 @@
+{% form_theme zoneForm 'PrestaShopBundle:Admin/TwigTemplateForm:prestashop_ui_kit.html.twig' %}
+
+{{ form_start(zoneForm) }}
+<div class="card">
```
need indentation :)

ks129 replied:
Other views don't have indent at there

PierreRambaud replied:
it is https://github.com/PrestaShop/PrestaShop/pull/20370/files#diff-cf4f05c9c12439e1bc28d4412266fbb3R30 :thinking:

ks129 replied:
https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Resources/views/Admin/Sell/Catalog/Categories/Blocks/form.html.twig#L29-L212 This don't have indent at there

NeOMakinG replied:
see form_start and form_end as a block, we don't have linter to check this, and maybe in the past peoples didn't care that much about codestyle while reviewing, but let's try to improve it a bit :)

PierreRambaud commented on src/PrestaShopBundle/Resources/views/Admin/Improve/International/Zone/edit.html.twig:
```diff
@@ -0,0 +1,36 @@
+{% set layoutTitle = 'Edit: %value%'|trans({'%value%': zoneName}, 'Admin.Actions') %}
+
+{% block content %}
+    <div class="row justify-content-center">
```
2 spaces like the others

PierreRambaud reviewed (dismissed):
Few comments :)

ks129 committed:
Fix sprintf type for presenting IDs for zones

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

ks129 committed:
Fix render path of zones index

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

ks129 committed:
Fix indent of zone edit view

ks129 committed:
Fix zones search path

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

LouiseBonnard commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,376 @@
+            ),
+            MissingZoneRequiredFieldsException::class => $this->trans(
+                'The field %s is required.',
+                'Admin.International.Notification',
```
```suggestion
                'Admin.Notifications.Error',
```

LouiseBonnard commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,376 @@
+            ),
+            ZoneNotFoundException::class => $this->trans(
+                'This zone does not exist.',
+                'Admin.International.Notification'
```
```suggestion
                'Admin.Notifications.Error'
```

atomiix commented on src/Core/Domain/Zone/ValueObject/ZoneId.php:
```diff
@@ -0,0 +1,69 @@
+     */
+    private function assertIntegerIsGreaterThanZero(int $zoneId)
+    {
+        if (0 > $zoneId) {
```
```suggestion
        if (0 >= $zoneId) {
```

ks129 committed:
Fix wrong translation domains

Co-authored-by: LouiseBonnard <32565890+LouiseBonnard@users.noreply.github.com>

ks129 committed:
Fix zone ID object constraint comparing

Co-authored-by: atomiix <tbaccelli@gmail.com>

NeOMakinG reviewed (dismissed):
Time for QA probably :D

ngodefroy commented:
Hello @ks129 ,

Could you resolve conflicts? 

Thank you

ks129 commented:
@ngodefroy Conflicts resolved

NeOMakinG commented:
Closed and open to run Travis again which was failing

Progi1984 commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,376 @@
+            return [];
+        }
+
+        foreach ($zoneIds as $i => $zoneId) {
+            $zoneIds[$i] = (int) $zoneId;
+        }
+
+        return $zoneIds;
```
```suggestion
        return array_map('intval', $zoneIds);
```

Progi1984 commented on src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml:
```diff
@@ -1180,3 +1180,11 @@ services:
+      arguments:
+        - '@translator'
+      tags:
+        - { name: form.type }
```
```suggestion
        - name: form.type
```

Progi1984 reviewed (commented):
Some comments

ks129 committed:
Simplify Zone IDs converting to ints

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Remove unnecessary { } from name: form.type for zones

Co-authored-by: Progi1984 <progi1984@gmail.com>

sarahdib commented:
@ks129 The controller isn't displayed
<img width="1278" alt="Capture d’écran 2020-09-29 à 10 35 08" src="https://user-images.githubusercontent.com/3034677/94533566-928f6b80-023f-11eb-8a88-c1605a876166.png">

ks129 committed:
Include Zones URLs to International URLs category

ks129 commented:
@sarahdib This issue should be fixed now.

ks129 committed:
Fix undefined method call

ks129 commented:
Conflicts resolved

khouloudbelguith commented:
Hi @ks129,

Could you please check conflicting files?

Thanks!

ks129 commented:
@khouloudbelguith Conflicts is now resolved.

Progi1984 commented on src/Core/Grid/Definition/Factory/ZoneGridDefinitionFactory.php:
```diff
@@ -0,0 +1,240 @@
+     */
+    private function getRowActions()
+    {
+        return (new RowActionCollection())
+            ->add(
+                (new LinkRowAction('edit'))
+                    ->setName($this->trans('Edit', [], 'Admin.Actions'))
+                    ->setIcon('edit')
+                    ->setOptions([
+                        'route' => 'admin_zones_edit',
+                        'route_param_name' => 'zoneId',
+                        'route_param_field' => 'id_zone',
+                        'clickable_row' => true,
+                    ])
+            )
+            ->add(
+                $this->buildDeleteAction(
+                    'admin_zones_delete',
+                    'zoneId',
+                    'id_zone'
+                )
+            );
```
```suggestion
         $rowActionCollection = new RowActionCollection();
         $rowActionCollection
           ->add(
                (new LinkRowAction('edit'))
                    ->setName($this->trans('Edit', [], 'Admin.Actions'))
                    ->setIcon('edit')
                    ->setOptions([
                        'route' => 'admin_zones_edit',
                        'route_param_name' => 'zoneId',
                        'route_param_field' => 'id_zone',
                        'clickable_row' => true,
                    ])
            )
            ->add(
                $this->buildDeleteAction(
                    'admin_zones_delete',
                    'zoneId',
                    'id_zone'
                )
            );
          return $rowActionCollection;
```

ks129 committed:
Fix Zones grid row actions getter return type

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Fix zones grid definition row action getter indention

ks129 committed:
Add missing newline

Arman-Hosseini commented on src/Adapter/Zone/CommandHandler/BulkToggleZoneStatusHandler.php:
```diff
@@ -0,0 +1,67 @@
+                if (!$zone->save()) {
+                    throw new CannotToggleZoneStatusException(sprintf('Unable to toggle status for zone with id "%d"', $zoneId->getValue()));
+                }
+            } catch (PrestaShopException $e) {
```
```suggestion
            } finally {
```

ks129 replied:
This doesn't make sense. This would throw exception every time, because `finally` is executed every time.

sarahdib commented:
@ks129  we can not access to zone 
<img width="1210" alt="Capture d’écran 2021-01-07 à 15 08 11" src="https://user-images.githubusercontent.com/3034677/103902012-3370ab00-50fa-11eb-96bc-7a065096e3f9.png">

ks129 committed:
Fix render template path of zones index

ks129 commented:
@sarahdib Issue is now fixed by fixing path of template.

sarahdib commented:
@ks129 
There is a bug with the search in the tab  : https://drive.google.com/file/d/1THEiuD2aZSSeaK1Hl6p3aLwKgebW868-/view

ks129 committed:
Fix search route wrong URL path

ks129 commented:
Problem is solved by changing URL path of search route back from `/search` to `/`.

sarahdib commented:
@ks129 

In 1.7.7.x using multishop there is an option to define for which shop the zone is created
<img width="1197" alt="Capture d’écran 2021-02-10 à 21 16 48" src="https://user-images.githubusercontent.com/3034677/107566726-8223d000-6be5-11eb-90e7-4f4e2c986593.png">

this option is missing with this Pull request
<img width="1221" alt="Capture d’écran 2021-02-10 à 21 17 03" src="https://user-images.githubusercontent.com/3034677/107566760-8fd95580-6be5-11eb-8a64-2d15d2a5fcd7.png">

If I create a shop only for shop A, I shouldn't see it in the shop B

ks129 committed:
Implement multistore to zones grid and form

Progi1984 reviewed (changes requested):
Need behat for CommandHandler & QueryHandler

ks129 committed:
Add zone behat tests

ks129 committed:
Remove unnecessary spaces

ks129 committed:
Fix zone behat tests

ks129 commented:
@Progi1984 Behat tests are now created.

Progi1984 commented on tests/Integration/Behaviour/Features/Scenario/Zone/zone_management.feature:
```diff
@@ -0,0 +1,46 @@
+    And I add new zone "jupiter" with following properties:
+      | name    | in-space  |
+      | enabled | false     |
+    When I enable multiple zones: "mars, jupiter" using bulk action
```
```suggestion
    Then zones: "mars, jupiter" should be disabled
    When I enable multiple zones: "mars, jupiter" using bulk action
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     *
+     * @param string $zoneReferences
+     */
+    public function assertZonesAreDeleted(string $zoneReferences)
```
```suggestion
    public function assertZonesAreDeleted(string $zoneReferences): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     *
+     * @param string $zoneReference
+     */
+    public function assertZoneIsDeleted(string $zoneReference)
```
```suggestion
    public function assertZoneIsDeleted(string $zoneReference): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $zoneReferences
+     * @param string $expectedStatus
+     */
+    public function assertMultipleZonesStatus(string $zoneReferences, string $expectedStatus)
```
```suggestion
    public function assertMultipleZonesStatus(string $zoneReferences, string $expectedStatus): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $zoneReference
+     * @param string $expectedStatus
+     */
+    public function assertStatus(string $zoneReference, string $expectedStatus)
```
```suggestion
    public function assertStatus(string $zoneReference, string $expectedStatus): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $zoneReference
+     * @param string $name
+     */
+    public function assertZoneName(string $zoneReference, string $name)
```
```suggestion
    public function assertZoneName(string $zoneReference, string $name): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $action
+     * @param string $zoneReferences
+     */
+    public function bulkToggleStatus(string $action, string $zoneReferences)
```
```suggestion
    public function bulkToggleStatus(string $action, string $zoneReferences): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     *
+     * @param string $zoneReference
+     */
+    public function toggleStatus(string $zoneReference)
```
```suggestion
    public function toggleStatus(string $zoneReference): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     *
+     * @param string $zoneReferences
+     */
+    public function bulkDeleteZones(string $zoneReferences)
```
```suggestion
    public function bulkDeleteZones(string $zoneReferences): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     *
+     * @param string $zoneReference
+     */
+    public function deleteZone(string $zoneReference)
```
```suggestion
    public function deleteZone(string $zoneReference): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $zoneReference
+     * @param TableNode $table
+     */
+    public function editZone(string $zoneReference, TableNode $table)
```
```suggestion
    public function editZone(string $zoneReference, TableNode $table): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     * @param string $zoneReference
+     * @param TableNode $table
+     */
+    public function createZone(string $zoneReference, TableNode $table)
```
```suggestion
    public function createZone(string $zoneReference, TableNode $table): void
```

Progi1984 commented on tests/Integration/Behaviour/Features/Context/Domain/ZoneFeatureContext.php:
```diff
@@ -0,0 +1,255 @@
+     */
+    private $defaultShopId;
+
+    public function __construct()
```
```suggestion
    public function __construct(): void
```

Progi1984 commented on src/Core/Domain/Zone/Command/BulkToggleZoneStatusCommand.php:
```diff
@@ -0,0 +1,79 @@
+    /**
+     * @param int[] $zoneIds
+     */
+    private function setZoneIds(array $zoneIds)
```
```suggestion
    private function setZoneIds(array $zoneIds): void
```

Progi1984 commented on src/Core/Domain/Zone/Command/BulkDeleteZoneCommand.php:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * @param int[] $zoneIds
+     */
+    private function setZoneIds(array $zoneIds)
```
```suggestion
    private function setZoneIds(array $zoneIds): void
```

Progi1984 commented on src/Core/Domain/Zone/Command/BulkToggleZoneStatusCommand.php:
```diff
@@ -0,0 +1,79 @@
+     */
+    private $zoneIds;
+
+    public function __construct(bool $expectedStatus, array $zoneIds)
```
```suggestion
    public function __construct(bool $expectedStatus, array $zoneIds): void
```

Progi1984 commented on src/Core/Domain/Zone/Command/BulkDeleteZoneCommand.php:
```diff
@@ -0,0 +1,68 @@
+    /**
+     * @param int[] $zoneIds
+     */
+    public function __construct(array $zoneIds)
```
```suggestion
    public function __construct(array $zoneIds): void
```

Progi1984 commented on src/Core/Domain/Zone/Command/AddZoneCommand.php:
```diff
@@ -0,0 +1,86 @@
+     * @param bool $enabled
+     * @param array $shopAssociation
+     */
+    public function __construct(string $name, bool $enabled, array $shopAssociation)
```
```suggestion
    public function __construct(string $name, bool $enabled, array $shopAssociation): void
```

Progi1984 commented on src/Core/Domain/Manufacturer/Command/ToggleManufacturerStatusCommand.php:
```diff
@@ -60,15 +60,15 @@ public function __construct($manufacturerId, $expectedStatus)
      * @return ManufacturerId
      */
-    public function getManufacturerId()
+    public function getManufacturerId(): ManufacturerId
```
```suggestion
    public function getManufacturerId()
```

It would be a BC

Progi1984 commented on src/Core/Domain/Manufacturer/Command/ToggleManufacturerStatusCommand.php:
```diff
@@ -60,15 +60,15 @@ public function __construct($manufacturerId, $expectedStatus)
      * @return bool
      */
-    public function getExpectedStatus()
+    public function getExpectedStatus(): bool
```
```suggestion
    public function getExpectedStatus()
```

It would be a BC

Progi1984 commented on src/Adapter/Zone/QueryHandler/GetZoneForEditingHandler.php:
```diff
@@ -0,0 +1,63 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(GetZoneForEditing $query)
```
```suggestion
    public function handle(GetZoneForEditing $query): EditableZone
```

Progi1984 commented on src/Core/Domain/Zone/QueryHandler/GetZoneForEditingHandlerInterface.php:
```diff
@@ -0,0 +1,45 @@
+     *
+     * @return EditableZone
+     */
+    public function handle(GetZoneForEditing $query);
```
```suggestion
    public function handle(GetZoneForEditing $query): EditableZone;
```

Progi1984 commented on src/Adapter/Zone/CommandHandler/AddZoneHandler.php:
```diff
@@ -0,0 +1,72 @@
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(AddZoneCommand $command)
```
```suggestion
    public function handle(AddZoneCommand $command): ZoneId
```

Progi1984 commented on src/Core/Domain/Zone/CommandHandler/AddZoneHandlerInterface.php:
```diff
@@ -0,0 +1,45 @@
+     *
+     * @return ZoneId
+     */
+    public function handle(AddZoneCommand $command);
```
```suggestion
    public function handle(AddZoneCommand $command): ZoneId;
```

Progi1984 reviewed (commented):
Some last comments (mainly type hinting) but it's a really good job.

ks129 committed:
Add return types to ZoneFeatureContext

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Remove BC changes that IDE automatically added

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Add return types to zone commands

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Add more return types to commands, queries and handlers

ks129 committed:
Test is zones disabled before mass toggling

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Remove constructor void return types

ks129 commented:
@Progi1984 Adding a `void` return type to constructors is not allowed and gives an error.

ks129 committed:
Remove void return type from ZoneFeatureContext constructor

Progi1984 commented:
> @Progi1984 Adding a `void` return type to constructors is not allowed and gives an error.

Ping @PierreRambaud FYI

Progi1984 commented on src/Adapter/Zone/CommandHandler/AddZoneHandler.php:
```diff
@@ -0,0 +1,72 @@
+            }
+
+            $this->associateWithShops($zone, $command->getShopAssociation());
+        } catch (\PrestaShopException $e) {
```
```suggestion
        } catch (PrestaShopException $e) {
```

Progi1984 commented on src/Adapter/Zone/CommandHandler/AddZoneHandler.php:
```diff
@@ -0,0 +1,72 @@
+use PrestaShop\PrestaShop\Core\Domain\Zone\CommandHandler\AddZoneHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Zone\Exception\MissingZoneRequiredFieldsException;
+use PrestaShop\PrestaShop\Core\Domain\Zone\Exception\ZoneException;
+use PrestaShop\PrestaShop\Core\Domain\Zone\ValueObject\ZoneId;
```
```suggestion
use PrestaShop\PrestaShop\Core\Domain\Zone\ValueObject\ZoneId;
use PrestaShopException;
```

Progi1984 commented on src/Core/Domain/Zone/CommandHandler/DeleteZoneHandlerInterface.php:
```diff
@@ -0,0 +1,42 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
```
```suggestion
```

Remove `declare(strict_types=1);` from interfaces, please

Progi1984 reviewed (commented):
Small comments

ks129 committed:
Use import instead direct reference

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Remove strict typing from interfaces

atomiix commented on src/Core/Grid/Query/ZoneQueryBuilder.php:
```diff
@@ -0,0 +1,126 @@
+                continue;
+            }
+
+            $builder->andWhere($filterName . ' LIKE :' . $filterName);
```
```suggestion
            $builder->andWhere('z.' . $filterName . ' LIKE :' . $filterName);
```

atomiix commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,372 @@
+        return $this->redirectToRoute('admin_zones_index');
+    }
+
+    private function getErrorMessages(Exception $e)
```
Can you add the phpdoc here?

atomiix commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,372 @@
+        ];
+    }
+
+    private function getBulkZonesFromRequest(Request $request)
```
Same here

atomiix commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/ZoneType.php:
```diff
@@ -0,0 +1,88 @@
+        $this->isMultistoreEnabled = $isMultistoreEnabled;
+    }
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
```
```suggestion
    /**
     * {@inheritdoc}
     */
    public function buildForm(FormBuilderInterface $builder, array $options)
```

ks129 committed:
Use alias prefix for filtering

Co-authored-by: atomiix <tbaccelli@gmail.com>

ks129 committed:
Add inheritdoc to buildForm

Co-authored-by: atomiix <tbaccelli@gmail.com>

ks129 committed:
Add PHPDoc and strict typing to ZoneController

khouloudbelguith commented:
Hi @ks129,

I have some issues:

**First issue**
1. In the BO > International > Locations > Zones list page
2. We can sort with the status `Enabled 
![image](https://user-images.githubusercontent.com/16067358/110324429-36aee700-8016-11eb-803a-7975e3c42699.png)
3. In the legacy page, it doesn't exist
![image](https://user-images.githubusercontent.com/16067358/110324535-678f1c00-8016-11eb-818b-22d20d9df4b5.png)

**Second issue**

1. In the BO > International > Locations > Zones list page
2. Try to filter by Zone: This filter is working only with the exact expression
3. If try to filter by  `Europe`, have only one result =>NOK
**With PR** 
![image](https://user-images.githubusercontent.com/16067358/110325114-3cf19300-8017-11eb-8764-c89107f8965b.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/110325224-5d215200-8017-11eb-8623-5250fa82c17d.png)

**Third issue**
1. Go to the BO > International > Locations > Zones page
2. Add new Zone
3. Set the Name with invalid data or too long data 
4. see error
`Invalid data`
**With PR **
![image](https://user-images.githubusercontent.com/16067358/110326187-9a3a1400-8018-11eb-942a-db99bb76771b.png)
**Without PR** 
![image](https://user-images.githubusercontent.com/16067358/110326274-b8a00f80-8018-11eb-828b-29c9b57cb050.png)

`Too long data`
**With PR**
![image](https://user-images.githubusercontent.com/16067358/110326349-d79ea180-8018-11eb-9116-65553f6a9b54.png)
**Without PR**
![image](https://user-images.githubusercontent.com/16067358/110326495-03ba2280-8019-11eb-9df8-23999fff6158.png)

 **Fourth issue**
1. Go to the BO > International > Locations > Zones page
2. Edit a Zone
3. Set the Name with invalid data or too long data 
4. see error 
![image](https://user-images.githubusercontent.com/16067358/110326647-3b28cf00-8019-11eb-9b4a-bfea96aa95b6.png)
![image](https://user-images.githubusercontent.com/16067358/110326690-4a0f8180-8019-11eb-8d99-c0cee65bf887.png)

**Fifth issue**
1. Enable Multistore context
2. Create a new shop
3. Go to the BO > International > Locations > Zones page
4. Select All shops
5. See error: All zones are duplicated 
![image](https://user-images.githubusercontent.com/16067358/110327061-c4d89c80-8019-11eb-9542-80acd131a242.png)


Thank you!

ks129 committed:
Make enabled field of zones grid not sortable

ks129 committed:
Wrap zone name between % for like search in grid

ks129 committed:
Add validators to zone form type

ks129 committed:
Fix multistore of zones grid

ks129 committed:
Change title of create zone page to Add new

ks129 committed:
Fix zone Mocha tests, what was broken because of migration

ks129 commented:
@khouloudbelguith All issues should be fixed now and also changed tests so they don't fail anymore.

ks129 committed:
Define visibility of zone delete error constants

ks129 committed:
Add missing return type to MissingZoneRequiredFieldsException

ks129 committed:
Add missing return types to ZoneId

ks129 committed:
Add missing return types to ZoneFormDataHandler

ks129 committed:
Add class docstring to ZoneFormDataHandler

ks129 committed:
Add missing return types to ZoneFormDataProvider

ks129 committed:
Add class docstring to ZoneFormDataProvider

ks129 committed:
Add missing return types to ZoneGridDefinitionFactory

ks129 committed:
Add missing return types to ZoneQueryBuilder

ks129 committed:
Add class docstring to ZoneQueryBuilder

boubkerbribri reviewed (dismissed):
LGTM for UI tests.
@khouloudbelguith , please re-run the tests for a double check.

ks129 committed:
Use inheritdoc for getData of ZoneFormDataProvider to avoid error

atomiix commented on src/Core/Domain/Zone/Exception/CannotToggleZoneStatusException.php:
```diff
@@ -0,0 +1,33 @@
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Core\Domain\Zone\Exception;
+
```
```suggestion

/**
 * Is thrown when cannot toggle zone.
 */
```

atomiix commented on src/Core/Grid/Query/ZoneQueryBuilder.php:
```diff
@@ -0,0 +1,130 @@
+     * @var int[]
+     */
+    private $contextShopIds;
+
```
```suggestion
    /**
     * @param Connection $connection
     * @param string $dbPrefix
     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
     * @param array $contextShopIds
     */
```

atomiix commented on src/Core/Grid/Query/ZoneQueryBuilder.php:
```diff
@@ -0,0 +1,130 @@
+    {
+        return $this->getQueryBuilder($searchCriteria)->select('COUNT(DISTINCT z.id_zone)');
+    }
+
```
```suggestion

    /**
     * @param SearchCriteriaInterface $criteria
     *
     * @return QueryBuilder
     */
```

atomiix reviewed (commented):
Only some phpdoc missing, you can ping me once done, I'll approve it right away ;)

ks129 committed:
Add missing PHPDoc

Co-authored-by: atomiix <tbaccelli@gmail.com>

ks129 commented:
@atomiix Changes made

ks129 committed:
Fix CS fixer

Progi1984 commented on src/Core/Form/IdentifiableObject/DataHandler/ZoneFormDataHandler.php:
```diff
@@ -0,0 +1,93 @@
+
+        if (isset($data['shop_association'])) {
+            $shopAssociation = $data['shop_association'] ?: [];
+            $shopAssociation = array_map(function ($shopId) { return (int) $shopId; }, $shopAssociation);
```
```suggestion
            $shopAssociation = array_map('intval', $shopAssociation);
```

Progi1984 commented on src/Core/Grid/Query/ZoneQueryBuilder.php:
```diff
@@ -0,0 +1,141 @@
+    }
+
+    /**
+     * @param SearchCriteriaInterface $criteria
```
```suggestion
     * @param SearchCriteriaInterface $searchCriteria
```

Progi1984 reviewed (commented):
Two small feedbacks and it's approved on my side ;)

ks129 committed:
Use intval instead of custom function

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 committed:
Fix PHPDoc wrong parameter name

Co-authored-by: Progi1984 <progi1984@gmail.com>

ks129 commented:
@Progi1984 Done

sowbiba commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+    }
+
+    /**
+     * Show "Add new" zone form and handles it's submit.
```
its

sowbiba replied:
You can ignore it :D :D

sowbiba reviewed (dismissed):
:ok_hand:

sarahdib commented:
Hello @ks129 

could you rebase ? 

Thank you

ks129 commented:
@sarahdib Conflicts is now resolved.

boubkerbribri reviewed (dismissed):
LGTM for UI tests.
No need to run them again.

NeOMakinG reviewed (dismissed):
Even if it has a little indentation discussion, the front-end LGTM to me

khouloudbelguith commented:
Hi @ks129, @boubkerbribri,

The automatic tests are ok :heavy_check_mark: 
[zone.pdf](https://github.com/PrestaShop/PrestaShop/files/6149295/zone.pdf)

But  @ks129, there is a new issue:
When we try to Add/Edit a zone, the Name value, the required symbol, and the Enabled value are missing 
![image](https://user-images.githubusercontent.com/16067358/111317744-b6ffc880-8664-11eb-9a95-e0e745f27622.png)

Thank you!

ks129 committed:
Add back (mysteriously) disappeared zone form labels

ks129 committed:
It's -> Its

ks129 committed:
Add intention to zone form

ks129 commented:
Conflicts is now resolved

PierreRambaud commented on src/Core/Domain/Zone/Command/BulkToggleZoneStatusCommand.php:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * @return ZoneId[]
```
you should prefer this annotation
```suggestion
     * @return array<int, ZoneId>
```

PierreRambaud commented on src/Core/Domain/Zone/Command/BulkToggleZoneStatusCommand.php:
```diff
@@ -0,0 +1,79 @@
+    private $expectedStatus;
+
+    /**
+     * @var ZoneId[]
```
```suggestion
     * @return array<int, ZoneId>
```

PierreRambaud commented on src/Core/Domain/Zone/Command/BulkToggleZoneStatusCommand.php:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * @param int[] $zoneIds
```
```suggestion
     * @param array<int, int> $zoneIds
```

PierreRambaud commented on src/Core/Form/IdentifiableObject/DataHandler/ZoneFormDataHandler.php:
```diff
@@ -0,0 +1,93 @@
+     */
+    public function create(array $data): int
+    {
+        if (!isset($data['shop_association']) || !$data['shop_association']) {
```
```suggestion
        if (empty($data['shop_association'])) {
```

PierreRambaud commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+        $zoneForm->handleRequest($request);
+
+        try {
+            $handleResult = $zoneFormHandler->handle($zoneForm);
```
The form should be check with isSubmitted and isValid?

ks129 replied:
I don't think so, at least other controller creation forms are not using this.

NeOMakinG replied:
@PierreRambaud this is the last feedback before QA probably, let's push it!

ks129 replied:
@PierreRambaud @NeOMakinG I think there is nothing to change. If you take a look at other controllers, there aren't also isSubmitted and isValid checks: these are only in edit forms. Here is one example:
https://github.com/PrestaShop/PrestaShop/blob/develop/src/PrestaShopBundle/Controller/Admin/Improve/International/TaxController.php#L153-L189

PierreRambaud commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+     */
+    public function deleteAction(int $zoneId): RedirectResponse
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteZoneCommand($zoneId));
+        } catch (ZoneException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));
+
+            return $this->redirectToRoute('admin_zones_index');
+        }
+        $this->addFlash('success', $this->trans('Successful deletion.', 'Admin.Notifications.Success'));
+
```
```suggestion
        try {
            $this->getCommandBus()->handle(new DeleteZoneCommand($zoneId));
            $this->addFlash('success', $this->trans('Successful deletion.', 'Admin.Notifications.Success'));
        } catch (ZoneException $e) {
            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages($e)));

            return $this->redirectToRoute('admin_zones_index');
        }

```

ks129 committed:
Use different PHPDoc annotation for zone commands

ks129 committed:
Simplify shop association check

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

ks129 committed:
Reorder actions in zone delete route

Co-authored-by: GoT <PierreRambaud@users.noreply.github.com>

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+            $handleResult = $zoneFormHandler->handle($zoneForm);
+
+            if (null !== $handleResult->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
```
```suggestion
                $this->addFlash('success', $this->trans('Successful creation', 'Admin.Notifications.Success'));
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteZoneCommand($zoneId));
+            $this->addFlash('success', $this->trans('Successful deletion.', 'Admin.Notifications.Success'));
```
```suggestion
            $this->addFlash('success', $this->trans('Successful deletion', 'Admin.Notifications.Success'));
```

Julievrz commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+                'Admin.International.Notification'
+            ),
+            MissingZoneRequiredFieldsException::class => $this->trans(
+                'The field %s is required.',
```
```suggestion
                'The %s field is required.',
```

Julievrz commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/ZoneType.php:
```diff
@@ -0,0 +1,114 @@
+                    new Length([
+                        'max' => 64,
+                        'maxMessage' => $this->translator->trans(
+                            'This field cannot be longer than %limit% characters',
```
```suggestion
                            'This field cannot be longer than %limit% characters.',
```

Julievrz commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/ZoneType.php:
```diff
@@ -0,0 +1,114 @@
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->translator->trans(
+                            'This field cannot be empty', [], 'Admin.Notifications.Error'
```
```suggestion
                            'This field cannot be empty.', [], 'Admin.Notifications.Error'
```

Julievrz commented on src/PrestaShopBundle/Form/Admin/Improve/International/Locations/ZoneType.php:
```diff
@@ -0,0 +1,114 @@
+                'constraints' => [
+                    new NotBlank([
+                        'message' => $this->translator->trans(
+                            'This field cannot be empty', [], 'Admin.Notifications.Error'
```
```suggestion
                            'This field cannot be empty.', [], 'Admin.Notifications.Error'
```

Julievrz reviewed (commented):
Hi, 
I've added some wording suggestions to improve the wording consistency with the rest of the software. Thanks!

ks129 committed:
Fix wording

Co-authored-by: Julie Varisellaz <70583503+Julievrz@users.noreply.github.com>

Julievrz reviewed (dismissed):
Thanks, the wording is ok.

NeOMakinG reviewed (dismissed):
Front and tests looks fine to me 👍 , @boubkerbribri or @nesrineabdmouleh could you take a look ? :)

boubkerbribri reviewed (dismissed):
@NeOMakinG Noting Changed on UI tests since my last review :slightly_smiling_face: .

Approved :heavy_check_mark:

ks129 commented:
This have got required approvals, shouldn't this move to QA?

NeOMakinG commented:
> This have got required approvals, shouldn't this move to QA?

Damn, it needs a rebase :(

NeOMakinG reviewed (dismissed):
Front-end LGTM

khouloudbelguith commented:
Hi @ks129, @PrestaShop/qa-automation,

The automatic tests are NOK.
If this PR is merged => Nightly will be NOK
[Zones.pdf](https://github.com/PrestaShop/PrestaShop/files/6228247/Zones.pdf)

Thanks!

boubkerbribri commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+            $handleResult = $zoneFormHandler->handle($zoneForm);
+
+            if (null !== $handleResult->getIdentifiableObjectId()) {
+                $this->addFlash('success', $this->trans('Successful creation', 'Admin.Notifications.Success'));
```
```suggestion
                $this->addFlash('success', $this->trans('Successful creation.', 'Admin.Notifications.Success'));
```

Julievrz replied:
Hi @boubkerbribri,
We try to use punctuation only on full sentences, e.g. 'The item was successfully deleted." but not on short sentences such as "Successful creation", "Successful deletion" that are not really full sentences. 
So I'd rather keep 'Successful creation" here. :-)

boubkerbribri replied:
Ok thank you

boubkerbribri commented on src/PrestaShopBundle/Controller/Admin/Improve/International/ZoneController.php:
```diff
@@ -0,0 +1,392 @@
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteZoneCommand($zoneId));
+            $this->addFlash('success', $this->trans('Successful deletion', 'Admin.Notifications.Success'));
```
```suggestion
            $this->addFlash('success', $this->trans('Successful deletion.', 'Admin.Notifications.Success'));
```

Julievrz replied:
I'd keep 'Successful deletion' without a period here.

boubkerbribri replied:
Thanks

boubkerbribri reviewed (commented):
@Julievrz could please review these suggestions.

@khouloudbelguith I think tests should be fixed after applying this.

Progi1984 commented:
@khouloudbelguith As wording is :green_circle: by @Julievrz, and automated tests will be fixed by @boubkerbribri after tests, it seems ok. Isn't it ?

khouloudbelguith commented:
Hi @Progi1984,

In fact, automated tests are NOK.
If we merge this PR, nightly will be broken.
They should be fixed in this PR I think.
Before the last commit, we have this translation `Successful creation.`
After the last commit, we have this translation 'Successful creation'
What do you think @boubkerbribri, tests should be updated?

Thank you!

boubkerbribri commented on tests/UI/pages/BO/international/locations/index.js:
```diff
@@ -6,71 +6,54 @@ class Zones extends BOBasePage {
 
     this.pageTitle = 'Zones •';
+    this.successfulUpdateStatusMessage = 'The status has been successfully updated.';
+    this.successfulUpdateMessage = 'Update successful';
```
```suggestion
    this.successfulCreationMessage = 'Successful creation';
    this.successfulUpdateMessage = 'Update successful';
    this.successfulDeleteMessage = 'Successful deletion';
```

Progi1984 committed:
Fixed UI tests

Co-authored-by: boubkerbribri <48441421+boubkerbribri@users.noreply.github.com>

Progi1984 commented:
@ks129 No problem.

Progi1984 commented:
@khouloudbelguith CI is :green_circle:

khouloudbelguith commented:
Hi @ks129,

It is ok :heavy_check_mark: 
https://drive.google.com/file/d/1AZ4TLQh3ENyDPsTTDbQViZnt4WnRK-zc/view?usp=sharing
tests are ok :heavy_check_mark: 
[zone_tests.pdf](https://github.com/PrestaShop/PrestaShop/files/6230314/zone_tests.pdf)

Thank you!

Progi1984 commented:
Thanks @ks129 & @khouloudbelguith
